### ISSUE TITLE
Inlining and verification in spec mode

### ISSUE THREAD

**Issue by hajduakos** [boogie-org/boogie#125]
I would like inlined methods to be also verified "on their own". The help displayed by the `/attrHelp` flag states that "methods with the attribute {:inline N} are not verified". However, using `/inline:spec` seems to both inline the method and verify it on its own.

Consider the following example:

```
var x : int;

procedure {:inline 1} inlinedproc()
    requires x >= 0;
    ensures x >= 0;
{
    x := x + 1;
}

procedure normalproc()
{
    x := 0;
    call inlinedproc();
    assert(x >= 1);
}
```

Running Boogie with `/doModSetAnalysis /inline:spec /trace` outputs the following:
```
Verifying inlinedproc ...
  [0.146 s, 1 proof obligation]  verified

Verifying normalproc ...
  [0.010 s, 3 proof obligations]  verified
```

There must be inlining, otherwise the assertion in `normalproc` could not be proved. However, the output indicates that `inlinedproc` was also verified on its own.

Is this a feature that can be relied on, or rather a bug? If it's a bug, creating a copy of the function without inlining could work, but it would be still nice to have it as a feature.

**Comment by shazqadeer** [comment]
I looked at the code and it looks like there are three inlining options: /inline:{none, assert, assume, spec}.  Verification is skipped only in the case of /inline:{assert, assume}. See the method SkipVerification in Absy.cs.  Your understanding of /inline:spec appears to be correct.

**Comment by hajduakos** [comment]
Issue solved by PR #126
